scripts/test-path-filters.sh

```
@@ -0,0 +1,309 @@
+        if test_filter "$filter" "${changed_files[@]}"; then
+            triggered_filters+=("$filter")
+        fi
+    done
+    
+    echo ""
+    
+    # Calculate optimization
+    local containers_to_build=0
+    
+    # Check if shared or critical files changed (affects all containers)
+    if [[ " ${triggered_filters[@]} " =~ " shared " ]] || \
```

Shellcheck (0.10.0)

[error] Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

(SC2199)